statescript/executor.go

```diff
@@ -296,10 +315,14 @@ func reportScriptStatus(rep *client.StatusReportWrapper, statusReport string) er
 
 func (l Launcher) ExecuteAll(state, action string, ignoreError bool,
 	report *client.StatusReportWrapper) error {
+	// Only execute the state-scripts if the version file is present.
+	if err := l.CheckRootfsScriptsVersion(); err != nil {
```

Comment:
No, this is not right. The rootfs version file only applies to the rootfs scripts (the ones without `Artifact` in the name), so it can be missing even if the artifact is using artifact scripts. So the check needs to happen later as it already does, when we know which type of script we need to run.